File: website/src/services/vehicle.ts
// Interfaces witht the API backend for vehicles 
export class VehicleApi {
    async getStaticVehicle() {
        try {
            const response = await fetch("https://localhost:5001/api/Vehicle/static", {
                credentials: 'include'
            });
            const json = await response.json();
            console.log(json);
        } catch (err) {
            console.log(err);
        }
    }

    async getAllVehicles(): Promise<Array<Vehicle>> {
        try {
            const response = await fetch(`https://localhost:5001/api/Vehicle/all`, {
                credentials: 'include'
            });
            const json = await response.json();
            console.log(json);

            return json;
        } catch (err) {
            console.log(err);

            return null;
        }
    }

    async getVehicles(num='10'): Promise<Array<Vehicle>> {
        try {
            const response = await fetch(`https://localhost:5001/api/Vehicle/${num}`, {
                credentials: 'include'
            });
            const json = await response.json();
            console.log(json);

            return json;
        } catch (err) {
            console.log(err);

            return null;
        }
    }
}

export interface Vehicle {
    id: string,
    make: string,
    model: string,
    trim: string,
    vin: string,
    year: Number,
}

File: api/api/Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace api.Models
{
    public class Vehicle
    {
        // "id" field from json
        public Guid Id { get; set; }

        public string Make { get; set; }

        public string Model { get; set; }

        public int Year { get; set; }

        public string Vin { get; set; }

        public string Trim { get; set; }
    }
}
